nautobot/extras/api/relationships.py

```diff
@@ -259,6 +256,49 @@ def to_internal_value(self, data):
         return {self.field_name: output_data}
 
 
+def api_relationships_errors(instance, relationships_data):
+    required_relationships = Relationship.objects.get_required_for_model(instance)
+    relationships_errors = []
+    for side, relations in required_relationships.items():
+        for relation in relations:
+
+            if relation.skip_required(instance, side):
+                continue
+
+            model_name = getattr(relation, f"{side}_type").model_class()._meta.verbose_name
+
+            try:
+                submitted_data = relationships_data.get(relation, "").get(side, "")
+            except AttributeError:
+                submitted_data = []
+
+            if len(submitted_data) == 0:
+                if relation.type in [
+                    RelationshipTypeChoices.TYPE_ONE_TO_MANY,
+                    RelationshipTypeChoices.TYPE_MANY_TO_MANY,
+                ]:
```

Comment:
Might be better to check `relation.has_many(side)` or `relation.has_many(RelationshipSideChoices.OPPOSITE[side])` as appropriate?